### PR TITLE
remove code that sometimes changed next_session_iso to first session.

### PR DESCRIPTION
**Pull request by darrellsilver**
Fixes a bug where next session time was appearing in wrong time zone when it was the student's first session. Affecting Myna, but Eagle needs qa too

### CODE REVIEW

**Comment by darrellsilver** [comment]
@Grae-Drake discussed w/ evan & bill; will put in qa so you guys can qa it in eagle + myna (elsewhere?) before it goes out.

File: models/Agreement.js [hunk 1]
```diff
@@ -19,28 +19,14 @@ Agreement.prototype.getFirstSession = function() {
...
 
 Agreement.prototype.getNextSession = function() {
-    // next_session is a fickle mistress, instead we'll use the day of the week
-    // and construct a new date
-    var session = moment(this.next_session_iso);
-    var current = moment();
-    var next;
-
-    if(!this.hasBegun()) {
-        next = this.getFirstSession();
```
Comment (evanrs) [inline models/Agreement.js]: It's almost impossible for me to resist harping on this one more time, but the fact that the pysplash model inconsistently represents timezones among its fields is quite simply some bogus cow dung.

As I mentioned before it improperly defines its relationships to other entities. In a less cow dung'y implementation it would be unambiguous by defining entity dependent fields separately.
```es6
class Agreement {
    __repr__() {
        return {
            members: [
                {
                    id: mentor.id,
                    role: mentor.role,
                    // in ISO, because why would you ever ever do otherwise?
                    first_session: mentor.first_session_iso_date,
                    next_session: mentor.next_session_iso_date,
                    timezone: mentor.timezone
                },
                {
                    id: student.id,
                    role: student.role,
                    first_session: student.first_session_iso_date,
                    next_session: student.next_session_iso_date,
                    timezone: student.timezone
                }
            ],
            ...agreementStuff
        }
    }
}
```

The idea that this duplicates data is incorrect. Our data is dependent, that means its the product of the timezone.
```
date != (date • profile.timezone)
```
It is a one way mutation that requires special knowledge to revert. We defer to the client code to enforce this differentiation—browser and server alike. It's a really unfortunate oversight that's hardly defensible. Again, to beat a now well worn (most likey dead) horse:
```
(first_session ÷ student.first_session) != (next_session ÷ student.next_session)
``` 
Therefore a well designed data model expresses these relations along side the dependent entity.
```es6
class Model {
    relationships: []
    mutation(relatedEntity) {
        relationships.append(join(relatedEntity, model.fields))
    }
}
```
TL;DR: I have feels, I have engineering feels.

**Comment by evanrs** [comment]
Other than my diatribe on good model design, :+1:

**Comment by Grae-Drake** [comment]
@darrellsilver is this ready to QA tomorrow?

**Comment by darrellsilver** [comment]
Tati was setting up a computer for it tonight - will send your way in the
morning for QA.

On Thu, Apr 30, 2015 at 9:31 PM, Grae-Drake <notifications@github.com>
wrote:

> @darrellsilver <https://github.com/darrellsilver> is this ready to QA
> tomorrow?
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/Thinkful/tfcommons.js/pull/8#issuecomment-98019287>.
>



-- 
Darrell Silver
Co-founder of Thinkful

On Twitter: http://twitter.com/darrellsilver
At work: http://blog.thinkful.com/

**Comment by Grae-Drake** [comment]
In Q/A process now. Initial testing looks good.  @varunrajan is doing followup testing and running into an issue with the Q/A environment.

Notably, this *does not* fix the issue we saw yesterday with system time affecting display times.  To reproduce, change your system time and navigate to edit a student session time (e.g. https://activity.thinkful.com/student/samuel-jaimes-5467/record/4159/create) or create a session record (e.g. https://activity.thinkful.com/student/samuel-jaimes-5467/record/4159/create).

**Comment by darrellsilver** [comment]
@Grae-Drake the issue ^ is resolved with https://github.com/Thinkful/myna/pull/190

Once this qa process is done let's release both.

**Comment by darrellsilver** [comment]
@Grae-Drake & @varunrajan please finish qa – I believe no open issues.

**Comment by Grae-Drake** [comment]
@darrellsilver how do we QA the new code?

**Comment by Grae-Drake** [comment]
Pay no attention to the man behind the curtain.

**Comment by varunrajan** [comment]
Editing the next session time updates it appropriately in the backend, but requires refreshing or leaving the page in order to display correctly.

Before refreshing, the displayed time takes the computer's time into account.

**Comment by varunrajan** [comment]
Also: after trying to delete an agreement -- student still shows up as active on the mentor's list.

**Comment by darrellsilver** [comment]
@varunrajan did you get a chance to open that "when deleting an agremeent... " issue?

**Comment by varunrajan** [comment]
@darrellsilver will do shortly -- assembling instances of related mentor grievances
